hive-metastore/src/test/java/org/apache/iceberg/hive/TestHiveMetastore.java

```diff
@@ -66,26 +72,35 @@
   private ExecutorService executorService;
   private TServer server;
   private HiveMetaStore.HMSHandler baseHandler;
+  private HiveClientPool clientPool;
 
   public void start() {
     try {
-      hiveLocalDir = createTempDirectory("hive", asFileAttribute(fromString("rwxrwxrwx"))).toFile();
+      this.hiveLocalDir = createTempDirectory("hive", asFileAttribute(fromString("rwxrwxrwx"))).toFile();
       File derbyLogFile = new File(hiveLocalDir, "derby.log");
       System.setProperty("derby.stream.error.file", derbyLogFile.getAbsolutePath());
       setupMetastoreDB("jdbc:derby:" + getDerbyPath() + ";create=true");
 
       TServerSocket socket = new TServerSocket(0);
       int port = socket.getServerSocket().getLocalPort();
-      hiveConf = newHiveConf(port);
-      server = newThriftServer(socket, hiveConf);
-      executorService = Executors.newSingleThreadExecutor();
-      executorService.submit(() -> server.serve());
+      this.hiveConf = newHiveConf(port);
+      this.server = newThriftServer(socket, hiveConf);
+      this.executorService = Executors.newSingleThreadExecutor();
+      this.executorService.submit(() -> server.serve());
+
+      // in Hive3, setting this as a system prop ensures that it will be picked up whenever a new HiveConf is created
+      System.setProperty(HiveConf.ConfVars.METASTOREURIS.varname, hiveConf.getVar(HiveConf.ConfVars.METASTOREURIS));
```

Comment:
Why is this needed? It may be nice to use this for applications, but for testing it seems like this allows us to pass the configuration incorrectly and still have tests pass. If it isn't needed, I'd rather remove it so that we must pass configuration properly.

Reply:
The Hive3 patch brought this in. If I understood @marton-bod correctly this was needed because in the HiveRunner there are some places where the config is not passed down, and this was the only way to fix the HMS client misery.
What I did here was only move this from the test class to the TestHiveMetastore class so it will affect the ClientPools as well

Reply:
That's correct. The rationale is outlined here: https://github.com/apache/iceberg/pull/1478#issuecomment-704961465
As you mentioned, when using Hive3, not all threads spawned during `HiveRunner` initialization receive our custom properties we set on the `HiveShell`, therefore using the system props is there to ensure they're picked up during `HiveConf` construction. This is needed to get the tests working properly on Hive3, without the `PersistenceManager`-related flaky tests.

Reply:
Okay, thanks for the reasoning. I guess I missed that this required a global setting in the other PR. I think this is fairly safe, since we always want to override the metastore URI using an option from the catalog's config.